infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationAckReader.java

```diff
@@ -471,7 +471,7 @@ public void run() {
                     try {
                         lock.lock();
                         long entriesToSend = calculateRemainingEntriesToSend(lastAckedTimestamp);
-                        metadataManager.setReplicationStatusTable(remoteClusterId, entriesToSend, lastSyncType);
+                        metadataManager.updateRemainingEntriesToSend(remoteClusterId, entriesToSend, lastSyncType);
```

Comment:
I agree with moving the lock outside of the try clause, and I think the only use for the locks is to avoid race conditions with the polling task but the polling task has been updated in this change to not modify anything but the remaining entries so there may be no use for the locks unless I am missing something.